ansible_runner/utils.py

```diff
@@ -82,6 +83,25 @@ def check_isolation_executable_installed(isolation_executable):
         return False
 
 
+def stream_dir(directory, **additional):
+    buf = BytesIO()
+    with zipfile.ZipFile(buf, 'w', compression=zipfile.ZIP_DEFLATED, allowZip64=True) as archive:
+        if directory:
+            for dirpath, dirs, files in os.walk(directory):
+                relpath = os.path.relpath(dirpath, directory)
+                if relpath == ".":
+                    relpath = ""
+                for fname in files:
+                    archive.write(os.path.join(dirpath, fname), arcname=os.path.join(relpath, fname))
```

Comment:
I think we have to be more selective here. I ran your demo commands, and found a surprisingly large zipfile was being sent. I turned out, it's because it was sending all my past artifacts, which are in the private_data_dir.